### PR TITLE
Adding loading with retry to workflow definition screen

### PR DESCRIPTION
**Pull request by tiagodolphine**
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/swf-backend/workflow-service/src/main/resources/swf1.svg
Comment (manstis) [inline plugins/swf-backend/workflow-service/src/main/resources/swf1.svg:1]: Any reason to include this file?

Comment (manstis) [inline plugins/swf-backend/workflow-service/src/main/resources/swf1.svg:1]: Any reason to include this file?

File: plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx [hunk 1]
```diff
@@ -74,12 +81,33 @@ export const SWFDefinitionViewerPage = () => {
     });
   }, []);
 
+  const errorCount = useRef(0);
```
Comment (manstis) [inline plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx]: Any reason to use `useRef` and not `useState`?

File: plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx [hunk 2]
```diff
@@ -74,12 +81,33 @@ export const SWFDefinitionViewerPage = () => {
...
+  const loading = useRef(true);
```
Comment (manstis) [inline plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx]: Likewise, can you please explain why you use `useRef`? I might simply not understand.

File: plugins/swf/src/components/SWFInstanceViewerPage/SWFInstanceViewerPage.tsx
Comment (manstis) [inline plugins/swf/src/components/SWFInstanceViewerPage/SWFInstanceViewerPage.tsx:1]: This page is not used anywhere and can be deleted.

I can do on a separate commit.

File: plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx [hunk 3]
```diff
@@ -74,12 +81,33 @@ export const SWFDefinitionViewerPage = () => {
...
+
+  const fetchData = useCallback(async () => {
+    swfApi
```
Comment (manstis) [inline plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx:81]: I think we should probably move the retry mechanism to the `swf-backend` plugin (or at least `SwfClient`).

Comment (tiagodolphine) [reply plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx:81]: makes sense, it is better, I'll change it

File: plugins/swf/src/components/SWFInstanceViewerPage/SWFInstanceViewerPage.tsx
Comment (tiagodolphine) [reply plugins/swf/src/components/SWFInstanceViewerPage/SWFInstanceViewerPage.tsx:1]: +1

File: plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx (hunk 2 above)
Comment (tiagodolphine) [reply plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx]: my lack of knowledge on react lol, I'll change it

File: plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx (hunk 1 above)
Comment (tiagodolphine) [reply plugins/swf/src/components/SWFDefinitionViewerPage/SWFDefinitionViewerPage.tsx]: my lack of knowledge on react lol, I'll change it

File: plugins/swf-backend/workflow-service/src/main/resources/swf1.svg
Comment (tiagodolphine) [reply plugins/swf-backend/workflow-service/src/main/resources/swf1.svg:1]: damm, it was generated by vscosde extension probably, we can remove it.

**Pull request closed without merging**